index.js

```diff
@@ -163,4 +163,12 @@ is.inRange = (x, range) => {
 
 is.infinite = x => x === Infinity || x === -Infinity;
 
+is.empty = value => {
```

Comment:
you've got quite a lot going on in this method.
1. get rid of unnecessary wrapping `( )`.
1. `value.length === 0` over `!value.length`
1. you basically have 3 assertions inside (aka `assertion1 || assertion2 || assertion3`). extract each assertion to a helper method (like `hasPromiseAPI`), and invoke them inside `is.empty`. This will also help identify which case each assertion relates to, and that will also help clarify which part of the method each assertion in the tests relates to.
1. what is the outcome of `is.empty(undefined)` or `is.empty(null)`? we need to make sure that all cases are taken care of, as the [README](https://github.com/sindresorhus/is#why-yet-another-type-checking-module) states,